graphql_compiler/tests/integration_tests/test_backends_integration.py

```diff
@@ -555,6 +555,112 @@ def test_recurse(self) -> None:
         for graphql_query, expected_results in queries:
             self.assertResultsEqual(graphql_query, parameters, test_backend.MSSQL, expected_results)
 
+    @integration_fixtures
+    def test_recurse_duplication_regression(self) -> None:
```

Comment:
Since this is quite a tricky test case, maybe add a docstring on the test function, or expand the comment on Query 4 to explain what's so special about it? Perhaps ideally also link to this PR, so we can easily link it back to the bug that we resolved and we are protecting against?